planet/src/main/webapp/js/placesItinerary.js

```diff
@@ -38,39 +41,75 @@ function initAutocomplete() {
     let options = {
         types: ['geocode']
     };
-    autocompleteStart = new google.maps.places.Autocomplete(startAddress,options); 
+    autocompleteStart = new google.maps.places.Autocomplete(startAddress,options);
+    autocompleteStart.setFields(['address_component']);
+    autocompleteStart.addListener('place_changed', updateStartingAddress);
+
     autocompleteEvent = new google.maps.places.Autocomplete(eventAddress,options);
 }
 
+function updateStartingAddress() {
+    const startAddress = document.getElementById('starting-address');
+    sessionStorage.setItem('start', startAddress.value);
+}
+
 export function renderPlaces() {
     displayedPlaces = {};
+    allPlaces = [];
+    placeCount = 0;
     const userId = firebase.auth().currentUser.uid;
     const placesRef = database.ref('users/' + userId + '/places');
-    placesRef.once('value', (placesSnapshot) => {
+    placesRef.orderByValue().once('value', (placesSnapshot) => {
         const placesContainer = document.getElementById('places');
         placesContainer.innerHTML = '';
         const savedPlaces = placesSnapshot.val();
         for (const placeId in savedPlaces) {
-            // Make request with fields
-            let placeRequest = {
-                placeId: placeId,
-                fields: ['place_id','name','rating','formatted_address',
-                    'opening_hours','photos','url']
-            };
-            let service = new google.maps.places.PlacesService(map);
-            service.getDetails(placeRequest, renderPlaceDetailsCallback); 
+            allPlaces.push(placeId);
         }
+        makePlaceRequests();
     });
 }
 
+function makePlaceRequests() {
+    hideLoadMoreButton();
+    let service = new google.maps.places.PlacesService(map);
+    for (let i = placeCount; i < placeCount + 10 ; i++) {
```

Comment:
10 is a magic number here. Prefer adding a CONSTANT above or a local const to indicate what this is.

https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants

Reply:
Fixed!